### PR TITLE
[MRG] GUI export configurations

### PR DESCRIPTION
**Pull request by gtdang**
Button to download network configurations to json.

### CODE REVIEW

File: hnn_core/hnn_io.py [hunk 1]
```diff
@@ -319,7 +320,7 @@ def network_to_dict(net, write_output=False):
 
 
 @fill_doc
```
Comment (ntolley) [inline hnn_core/hnn_io.py:322]:
```suggestion
```

File: hnn_core/hnn_io.py [hunk 2]
```diff
@@ -328,25 +329,33 @@ def write_network_configuration(net, fname, overwrite=True):
...
-
     net_data = net.to_dict(write_output=False)
     net_data_converted = _convert_np_array_to_list(net_data)
 
-    # Saving file
-    with open(fname, 'w', encoding='utf-8') as f:
-        json.dump(net_data_converted, f, ensure_ascii=False, indent=4)
+    if isinstance(output, (str, Path)):
+        if overwrite is False and os.path.exists(output):
+            raise FileExistsError('File already exists at path %s. Rename '
```
Comment (ntolley) [inline hnn_core/hnn_io.py:349]: Unit test for this warning?

File: hnn_core/hnn_io.py [hunk 3]
```diff
@@ -328,25 +329,33 @@ def write_network_configuration(net, fname, overwrite=True):
...
+                                  'the file or set overwrite=True.' % (output,)
+                                  )
+        # Saving file
+        with open(output, 'w', encoding='utf-8') as f:
+            json.dump(net_data_converted, f, ensure_ascii=False, indent=4)
+    else:
+        # Write to StringIO buffer
```
Comment (ntolley) [inline hnn_core/hnn_io.py:356]:
```suggestion
        # Write to StringIO buffer, used for GUI to store network config in memory
```

File: hnn_core/hnn_io.py [hunk 4]
```diff
@@ -328,25 +329,33 @@ def write_network_configuration(net, fname, overwrite=True):
...
 
-    # Saving file
-    with open(fname, 'w', encoding='utf-8') as f:
-        json.dump(net_data_converted, f, ensure_ascii=False, indent=4)
+    if isinstance(output, (str, Path)):
+        if overwrite is False and os.path.exists(output):
+            raise FileExistsError('File already exists at path %s. Rename '
+                                  'the file or set overwrite=True.' % (output,)
+                                  )
+        # Saving file
```
Comment (ntolley) [inline hnn_core/hnn_io.py:352]:
```suggestion
        # Saving file, used by API to directly export network config to file
```

**Comment by ntolley** [comment]
I'm happy with the changes here! Really big step getting this merged @gtdang, super excited to have this distributed in the next release

File: hnn_core/hnn_io.py (hunk 2 above)
Comment (gtdang) [reply hnn_core/hnn_io.py:349]: Yep! There's a test for this in test_io.

**Pull request merged**